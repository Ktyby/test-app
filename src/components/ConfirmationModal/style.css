.confirmation-modal__content {
  margin-bottom: 15px;
}

.confirmation-modal__text {
  margin: 0;

  font-size: 16px;
  text-align: center;
}

.confirmation-modal__text:not(:last-child) {
  margin-bottom: 5px;
}

.confirmation-modal__buttons {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.confirmation-modal__button {
  width: 50px;
  height: 30px;

  background-color: #fcd911;

  border: none;
  border-radius: 10px;

  font-size: 14px;

  cursor: pointer;

  transition: 0.4s;
}

.confirmation-modal__button:not(:last-child) {
  margin-right: 10px;
}

.confirmation-modal__button:hover {
  background-color: #000;
  color: #fff;
}
