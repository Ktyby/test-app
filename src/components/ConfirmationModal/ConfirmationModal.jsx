import React from "react";
import PropTypes from "prop-types";

import Modal from "../Modal";

import "./style.css";

const ConfirmationModal = ({ onDeleteUser, onClose, title }) => (
  <Modal className="confirmation-modal" title={title} onClose={onClose}>
    <div className="confirmation-modal__content">
      <p className="confirmation-modal__text">Are You sure?</p>
    </div>
    <div className="confirmation-modal__buttons">
      <button className="confirmation-modal__button" onClick={onDeleteUser}>
        Yep
      </button>
      <button className="confirmation-modal__button" onClick={onClose}>
        Nope
      </button>
    </div>
  </Modal>
);

ConfirmationModal.propTypes = {
  title: PropTypes.string,
  onClose: PropTypes.func,
  onDeleteUser: PropTypes.func,
};

ConfirmationModal.defaultProps = {
  title: "Confirmation",
  onClose: () => {},
  onDeleteUser: () => {},
};

export default ConfirmationModal;
