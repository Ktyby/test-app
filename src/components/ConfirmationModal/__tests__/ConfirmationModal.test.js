import renderer from "react-test-renderer";
import React from "react";

import ConfirmationModal from "../ConfirmationModal";

describe("components => ConfirmationModal", () => {
  it("should render ConfirmationModal component", () => {
    const component = renderer.create(<ConfirmationModal />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
