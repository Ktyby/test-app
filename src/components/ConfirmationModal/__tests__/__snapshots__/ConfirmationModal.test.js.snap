// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components => ConfirmationModal should render ConfirmationModal component 1`] = `
<div
  className="overlay"
  onClick={[Function]}
>
  <div
    className="modal"
  >
    <div
      className="modal__heading"
    >
      <h3
        className="modal__title"
      >
        Confirmation
      </h3>
      <button
        className="modal__close"
        onClick={[Function]}
      >
        â•³
      </button>
    </div>
    <div
      className="confirmation-modal__content"
    >
      <p
        className="confirmation-modal__text"
      >
        Are You sure?
      </p>
    </div>
    <div
      className="confirmation-modal__buttons"
    >
      <button
        className="confirmation-modal__button"
        onClick={[Function]}
      >
        Yep
      </button>
      <button
        className="confirmation-modal__button"
        onClick={[Function]}
      >
        Nope
      </button>
    </div>
  </div>
</div>
`;
